lib/lte_link_control/lte_lc.c

```diff
@@ -123,6 +123,15 @@ static const char system_mode_preference[] = {
 	[LTE_LC_SYSTEM_MODE_PREFER_NBIOT_PLMN_PRIO]	= '4',
 };
 
+static void lte_lc_psm_get_work_fn(struct k_work *work_item);
+K_WORK_DEFINE(lte_lc_psm_get_work, lte_lc_psm_get_work_fn);
+
+#define LTE_LC_PSM_GET_STACK_SIZE 1024
+#define LTE_LC_PSM_GET_PRIORITY 5
+K_THREAD_STACK_DEFINE(lte_lc_psm_get_stack, LTE_LC_PSM_GET_STACK_SIZE);
+
+static struct k_work_q lte_lc_psm_get_work_q;
```

Comment:
Right now I'm adding this work queue just for doing AT%XMONITOR when CEREG notification had been received. It's a bit awkward that we need to make a work queue and reserve memory for it for such a small task that occurs in a corner case but there's really no other way out.

We could think of sending all events from the work queue and even parsing AT notifications in the future.

We need to think if we want to delay this feature beyond 2.6.0 to mitigate the risks.

Reply:
IMHO, the work queue could be already named for example `lte_lc_work_q`, even though it's (for now) used only for this one use case.

Reply:
I would rename the work queue when we use it for something else. Because it may be that for the next two years, we won't be using it for anything else 😄

Reply:
I would still prefer the generic naming, but I'll leave it up to you to decide. 😄